libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2211,12 +2211,13 @@ bool TypeChecker::visit(IndexAccess const& _access)
 		else
 		{
 			expectType(*index, IntegerType(256));
-			if (auto numberType = dynamic_cast<RationalNumberType const*>(type(*index).get()))
-			{
-				if (!numberType->isFractional()) // error is reported above
-					if (!actualType.isDynamicallySized() && actualType.length() <= numberType->literalValue(nullptr))
-						m_errorReporter.typeError(_access.location(), "Out of bounds array access.");
-			}
+			if (!m_errorReporter.hasErrors())
+				if (auto numberType = dynamic_cast<RationalNumberType const*>(type(*index).get()))
+				{
+					if (!numberType->isFractional()) // error is reported above
```

Comment:
Since you now check for `hasErrors()` this should be changed into `solAssert(!numberType->isFractional(), "")`